WooCommerce/Classes/ViewRelated/Orders/Simple Payments/Amount/SimplePaymentsAmount.swift

```diff
@@ -126,10 +129,12 @@ struct SimplePaymentsAmount: View {
         .navigationTitle(Localization.title)
         .toolbar {
             ToolbarItem(placement: .cancellationAction) {
+                // Cancel button set to disabled state until get the result from `createSimplePaymentsOrder`
                 Button(Localization.cancelTitle, action: {
                     dismiss()
                     viewModel.userDidCancelFlow()
                 })
+                    .disabled(viewModel.loading)
```

Comment:
Ideally, we don't want the view to make assumptions about the view model state. Here we are assuming that we want the cancel button disabled when the view model is loading, and while that is the intended behavior a much more clear approach is to expose a `disableCancelButton` property on the view model that the view can consume without any doubt or assumption. Also, that allows us to unit test that new logic.

Does that make sense? What do you think?

Reply:
You are right, I will remember to write logics in `viewModel`s rather than in the view from now on. Thanks for pointing out.

Reply:
Added `disableCancelButton` property in `ViewModel`, that is using in `SimplePaymentsAmount` view to enable/disable cancel button.

```
Button(Localization.cancelTitle, action: {
                    dismiss()
                    viewModel.userDidCancelFlow()
                })
                    .disabled(viewModel.disableCancelButton())
```

Reply:
Unit test added for `disableCancelButton`.